packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useCallback, useEffect } from 'react';
+import { useBooleanState } from '../../lib/hooks';
+import { injectStripe, CardElement, ReactStripeElements } from 'react-stripe-elements';
+import { UpdatePaymentFetchState, CustomerFetchState } from '../../store/types';
+
+type PaymentUpdateFormProps = {
+  accessToken: string,
+  customer: CustomerFetchState,
+  resetUpdatePayment: Function,
+  updatePayment: Function,
+  updatePaymentStatus: UpdatePaymentFetchState
+};
+export const PaymentUpdateForm = ({
+  accessToken,
+  updatePayment,
+  updatePaymentStatus,
+  resetUpdatePayment,
+  customer,
+  stripe
+}: PaymentUpdateFormProps & ReactStripeElements.InjectedStripeProps) => {
+  const [ updateRevealed, revealUpdate, hideUpdate ] = useBooleanState();
+
+  // Reset payment update status on initial render.
+  useEffect(() => { 
+    resetUpdatePayment(); 
+  }, [ resetUpdatePayment ]);
+
+  const onSubmit = useCallback(ev => {
+    ev.preventDefault();
+
+    // TODO: use react state on form fields along with validation
+    const data = new FormData(ev.target);
+    const name = String(data.get('name'));
+
+    if (stripe) {
+      stripe
+        .createToken({ name })
+        .then((result) => {
+          updatePayment(accessToken, {
+            paymentToken: result && result.token && result.token.id,
+          });
+          hideUpdate(ev);
+        });
+        // TODO: error handling
+    }
+  }, [ accessToken, updatePayment, stripe ]);
+
+  if (customer.loading) {
+    // If the customer details are loading, then we have nothing to update yet.
+    return <span></span>;
+  }
+
+  if (customer.error) {
+    // If there's an error fetching the customer, there are no billing details to update.
+    // TODO: Specifically 404 error means no details, 401 / 500 could be reported differently.
+    return <span></span>;
+  }
+
+  if (updatePaymentStatus.loading) {
+    return (
+      <div>
+        <h3>Billing information</h3>
+        <p>Updating...</p>
+      </div>
+    );
+  }
+
+  if (updatePaymentStatus.error) {
+    return (
+      <div>
+        <h3>Billing information</h3>
+        <p>Updating... Error! {'' + updatePaymentStatus.error}</p>
+      </div>
+    );
+  }
+
+  const { payment_type, last4, exp_month, exp_year } = customer.result;
+  return (
+    <div>
+      <h3>Billing information</h3>
+      {/* TODO: TBD on UX for reporting payment update success */}
```

Comment:
In the Invision spec there's a topbar:

<img width="1464" alt="image" src="https://user-images.githubusercontent.com/44390/57724690-059b4080-7651-11e9-9614-ca646e3da454.png">

Reply:
Oh hey, I missed that - I'll add a portal and a bit of state to pop something up there.